### PR TITLE
resource_manager: persist settings with settings prefix

### PR DESCRIPTION
**Pull request by nolouch**
Signed-off-by: nolouch <nolouch@gmail.com>

### What problem does this PR solve?

Issue Number: Ref https://github.com/tikv/pd/issues/5784

### What is changed and how does it work?

```commit-message
resource_manager: persist settings with settings prefix
```

### Check List

Tests

- Unit test
- Integration test

```release-note
None
```

### CODE REVIEW

File: pkg/mcs/resource_manager/server/resource_group.go [hunk 1]
```diff
@@ -182,28 +189,33 @@ func (rg *ResourceGroup) IntoProtoResourceGroup() *rmpb.ResourceGroup {
...
 	}
 	return nil
 }
+
+// persistSettings persists the resource group settings.
+// TODO: persist the state of the group separately.
+func (rg *ResourceGroup) persistSettings(storage storage.Storage) error {
+	rg.RLock()
+	defer rg.RUnlock()
+	metaGroup := rg.IntoProtoResourceGroup()
```
Comment (CabinfeverB) [inline pkg/mcs/resource_manager/server/resource_group.go:219]: There is RLock in `IntoProtoResourceGroup `

**Review by CabinfeverB — APPROVED** [review]

**Comment by nolouch** [comment]
PTAL @JmPotato

File: pkg/mcs/resource_manager/server/resource_group.go [hunk 2]
```diff
@@ -17,14 +17,21 @@ package server
...
 	"github.com/pingcap/errors"
 	rmpb "github.com/pingcap/kvproto/pkg/resource_manager"
 	"github.com/pingcap/log"
+	"github.com/tikv/pd/server/storage"
 	"go.uber.org/zap"
 )
 
+const (
+	// GroupSettingsPathPrefix is the prefix of the resource group path to store group settings.
+	GroupSettingsPathPrefix = "/settings"
```
Comment (JmPotato) [inline pkg/mcs/resource_manager/server/resource_group.go:32]: It seems that this const is no need to become exportable.

File: pkg/mcs/resource_manager/server/resource_group.go [hunk 3]
```diff
@@ -92,12 +99,12 @@ func (rg *ResourceGroup) CheckAndInit() error {
...
 		}
 	}
-	if rg.Mode == rmpb.GroupMode_NativeMode {
+	if rg.Mode == rmpb.GroupMode_RawMode {
 		if rg.ResourceSettings == nil {
 			rg.ResourceSettings = &NativeResourceSettings{}
 		}
 		if rg.RUSettings != nil {
-			return errors.New("invalid resource group settings, native mode should not set RU settings")
+			return errors.New("invalid resource group settings, Raw mode should not set RU settings")
```
Comment (JmPotato) [inline pkg/mcs/resource_manager/server/resource_group.go:107]:
```suggestion
			return errors.New("invalid resource group settings, raw mode should not set RU settings")
```

File: pkg/mcs/resource_manager/server/resource_group.go [hunk 4]
```diff
@@ -106,26 +113,26 @@ func (rg *ResourceGroup) CheckAndInit() error {
...
-		rg.RUSettings.RRU.patch(groupSettings.GetRUSettings().GetRRU())
-		rg.RUSettings.WRU.patch(groupSettings.GetRUSettings().GetWRU())
-	case rmpb.GroupMode_NativeMode:
-		if groupSettings.GetResourceSettings() == nil {
-			return errors.New("invalid resource group settings, native mode should set resource settings")
+		rg.RUSettings.RRU.patch(metaGroup.GetRUSettings().GetRRU())
+		rg.RUSettings.WRU.patch(metaGroup.GetRUSettings().GetWRU())
+	case rmpb.GroupMode_RawMode:
+		if metaGroup.GetResourceSettings() == nil {
+			return errors.New("invalid resource group settings, Raw mode should set resource settings")
```
Comment (JmPotato) [inline pkg/mcs/resource_manager/server/resource_group.go:131]:
```suggestion
			return errors.New("invalid resource group settings, raw mode should set resource settings")
```

**Review by JmPotato — APPROVED** [review]

**Pull request merged**